pkg/3scale/amp/operator/system_mysql_options_provider.go

```diff
@@ -159,11 +159,22 @@ func (s *SystemMysqlOptionsProvider) setResourceRequirementsOptions() {
 }
 
 func (s *SystemMysqlOptionsProvider) setPersistentVolumeClaimOptions() {
+	var volumeName *string
+	storageRequests := component.DefaultSystemMysqlStorageResources()
+
 	if s.apimanager.Spec.System.DatabaseSpec != nil &&
 		s.apimanager.Spec.System.DatabaseSpec.MySQL != nil &&
 		s.apimanager.Spec.System.DatabaseSpec.MySQL.PersistentVolumeClaimSpec != nil {
+
 		s.mysqlOptions.PVCStorageClass = s.apimanager.Spec.System.DatabaseSpec.MySQL.PersistentVolumeClaimSpec.StorageClassName
+		volumeName = s.apimanager.Spec.System.DatabaseSpec.MySQL.PersistentVolumeClaimSpec.VolumeName
+		if s.apimanager.Spec.System.DatabaseSpec.MySQL.PersistentVolumeClaimSpec.Resources != nil {
+			storageRequests = s.apimanager.Spec.System.DatabaseSpec.MySQL.PersistentVolumeClaimSpec.Resources.Requests
+		}
 	}
+
+	s.mysqlOptions.PVCVolumeName = volumeName
```

Comment:
Can we move this one inside the conditional? Just assigning the value it has there, without the need to create the auxiliar variable either.

Reply:
what is the benefit? The way it is  now, `s.mysqlOptions.PVCVolumeName` is always assigned. The way you propose, it is only assigned when the conditional is true. Both would work, assuming no other method modifies that variable, but for me it is much clearer like this.

Reply:
To me it is more clear and less verbose the other way around. No need to define another variable where the zero-value does it already, as it's often done in Go. You only set it in case MySQL PVC section is defined, as with all the other parameters.

In any case, as you prefer, it's a matter of style.

Reply:
Let it be like this, then. This method `setPersistentVolumeClaimOptions` will decide the final value of `s.mysqlOptions.PVCVolumeName` without assuming its current value is zero value (nil).